packages/react-components/src/store/highlightedDataStreamsSlice.ts

```diff
@@ -0,0 +1,28 @@
+import { DataStream } from '@iot-app-kit/core';
+import { StateCreator } from 'zustand';
+
+export interface HighlightedDataStreamsData {
+  highlightedDataStreams: DataStream[];
+}
+
+export interface HighlightedDataSteamsState extends HighlightedDataStreamsData {
+  highlightDataStream: (datastream?: DataStream) => void;
+  unHighlightDataStream: (datastream?: DataStream) => void;
+  isDataStreamHighlighted: (datastream?: DataStream) => boolean;
+}
+
+export const createHighlightedDataStreamsSlice: StateCreator<HighlightedDataSteamsState> = (set, get) => ({
+  highlightedDataStreams: [],
+  highlightDataStream: (datastream?: DataStream) =>
+    set((state) => {
+      if (!datastream) return { highlightedDataStreams: state.highlightedDataStreams };
```

Comment:
you can just update only when if state is true ? OR may be the action is not even called if the datastreams is undefined ? 
i.e make `highlightDataStream: (datastream: DataStream)`